docker/ingress/build-nginx.sh

```diff
@@ -18,6 +18,25 @@ apt-get install --no-install-suggests --no-install-recommends -y \
 # allow nginx user to manage interfaces and arp configuration
 echo "%nginx ALL=NOPASSWD: /sbin/ip, /usr/bin/tee" >> /etc/sudoers
 
+echo "--- Downloading openSSL"
+mkdir /tmp/openssl
+cd /tmp/openssl
+OPENSSL="1_1_0g"
+OPENSSL_SHA256="8e9516b8635bb9113c51a7b5b27f9027692a56b104e75b709e588c3ffd6a0422"
+OPENSSL_tarball="OpenSSL_${OPENSSL}.tar.gz"
+
+curl -L -O https://github.com/openssl/openssl/archive/${OPENSSL_tarball}
+
+touch hashes_openssl
+echo "${OPENSSL_SHA256} ${OPENSSL_tarball}" >> hashes_openssl
+if ! sha256sum -c hashes_openssl; then
+    echo "sha256 hashes do not match downloaded files"
+    exit 1
+fi
+
+tar xzf ${OPENSSL_tarball}
+mv openssl-OpenSSL_${OPENSSL} openssl
+
```

Comment:
How come we're downloading the tarball? Why not install with apt, so we can make sure we're getting the latest bugfix on ubuntu?

Reply:
We're also going to need some sort of check mechanism to make sure we keep this up to date...

Reply:
Exactly the same issue for nginx build, and ubuntu didn't update openssl fast enough when you have a CVE.
I wonder if we can use libreSSL instead, seems much better without all the CVE of openssl.